boa/src/vm/mod.rs

```diff
@@ -5,25 +5,44 @@ pub(crate) mod instructions;
 
 pub use compilation::Compiler;
 pub use instructions::Instruction;
+use std::time::{Duration, Instant};
 
 // Virtual Machine.
 #[derive(Debug)]
 pub struct VM<'a> {
     ctx: &'a mut Context,
+    idx: usize,
     instructions: Vec<Instruction>,
     pool: Vec<Value>,
     stack: Vec<Value>,
     stack_pointer: usize,
+    profile: Profiler,
+    is_trace: bool,
+}
+#[derive(Debug)]
+pub struct Profiler {
```

Comment:
I would add some space here between the two definitions, and add some documentation to for example explain the difference between this profiler and the general Boa profiler.

Also, do we need to modify the string below or the vectors above? If they are constant once we initialize the structure, it would be best to use `Box<[Value]>` and `Box<str>` for vectors and strings respectively, for some memory usage improvements.